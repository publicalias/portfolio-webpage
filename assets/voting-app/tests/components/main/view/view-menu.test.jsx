"use strict";

//local imports

const ViewMenu = require("../../../../scripts/components/main/view/view-menu");

const { testWrapper } = require("../../../test-helpers");

//global imports

const { newUser } = require("schemas/master");
const { newPoll } = require("schemas/voting-app");
const { initTestClick, reactTests } = require("test-helpers/react-tests");
const { deepCopy } = require("utilities");

//setup

beforeAll(reactTests.setup);

//view menu

describe("view menu", () => {

  const { testMount, testShallow } = testWrapper(ViewMenu);

  const props = [{ user: newUser({ id: "id-a" }) }, { poll: newPoll({ users: { created: "id-a" } }) }];

  const testClick = initTestClick(testMount);

  it("should match snapshot (default)", () => {

    const { wrapper } = testShallow({}, { poll: newPoll() });

    expect(wrapper).toMatchSnapshot();

  });

  it("should match snapshot (created)", () => {

    const { wrapper } = testShallow(...props);

    expect(wrapper).toMatchSnapshot();

  });

  it("should match snapshot (created, settings)", () => {

    const localProps = deepCopy(props, [{ view: { settings: true } }]);

    const { wrapper } = testShallow(...localProps);

    expect(wrapper).toMatchSnapshot();

  });

  it("should call viewToggleSettings on click", () => testClick(".qa-toggle-settings", "viewToggleSettings", [], ...props));

});
