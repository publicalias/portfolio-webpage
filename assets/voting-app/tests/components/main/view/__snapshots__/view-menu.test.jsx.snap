// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view menu should match snapshot (created) 1`] = `
<div
  className="c-view-menu"
>
  <button
    className="c-view-menu__control-btn qa-toggle-settings"
    onClick={[Function]}
  >
    Settings
  </button>
  <ViewControls
    actions={
      Object {
        "formAddOption": [MockFunction],
        "formClearState": [MockFunction],
        "formRemoveOption": [MockFunction],
        "formSetAdd": [MockFunction],
        "formSetTitle": [MockFunction],
        "formToggleConfirm": [MockFunction],
        "formToggleSecret": [MockFunction],
        "listClearState": [MockFunction],
        "listSetSearch": [MockFunction],
        "metaAddErrors": [MockFunction],
        "metaCloseError": [MockFunction],
        "metaCreatePoll": [MockFunction],
        "metaDeletePoll": [MockFunction],
        "metaGetPolls": [MockFunction],
        "metaGetUser": [MockFunction],
        "metaNoOp": [MockFunction],
        "metaSetLoading": [MockFunction],
        "metaSetState": [MockFunction],
        "metaTimeoutError": [MockFunction],
        "pollAddOption": [MockFunction],
        "pollCastVote": [MockFunction],
        "pollRemoveOption": [MockFunction],
        "pollToggleFlag": [MockFunction],
        "pollToggleHide": [MockFunction],
        "pollToggleSecret": [MockFunction],
        "viewClearState": [MockFunction],
        "viewSetAdd": [MockFunction],
        "viewToggleConfirm": [MockFunction],
        "viewToggleSettings": [MockFunction],
      }
    }
    data={
      Object {
        "errors": Array [],
        "form": Object {
          "add": "",
          "confirm": false,
          "options": Array [],
          "secret": false,
          "title": "",
        },
        "list": Object {
          "search": "",
        },
        "loading": false,
        "polls": Array [],
        "user": Object {
          "auth": Object {
            "id": "",
            "provider": "",
          },
          "data": Object {
            "restricted": false,
          },
          "id": "id-a",
          "name": "",
          "type": "auth",
        },
        "view": Object {
          "add": "",
          "confirm": false,
          "settings": false,
        },
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    location={Object {}}
    match={Object {}}
    poll={
      Object {
        "author": "",
        "date": 0,
        "id": "",
        "options": Array [],
        "secret": false,
        "title": "",
        "users": Object {
          "created": "id-a",
          "flagged": Array [],
          "hidden": Array [],
          "voted": 0,
        },
      }
    }
    staticContext={Object {}}
  />
</div>
`;

exports[`view menu should match snapshot (created, settings) 1`] = `
<div
  className="c-view-menu"
>
  <button
    className="c-view-menu__control-btn qa-toggle-settings"
    onClick={[Function]}
  >
    Settings
  </button>
  <ViewSettings
    actions={
      Object {
        "formAddOption": [MockFunction],
        "formClearState": [MockFunction],
        "formRemoveOption": [MockFunction],
        "formSetAdd": [MockFunction],
        "formSetTitle": [MockFunction],
        "formToggleConfirm": [MockFunction],
        "formToggleSecret": [MockFunction],
        "listClearState": [MockFunction],
        "listSetSearch": [MockFunction],
        "metaAddErrors": [MockFunction],
        "metaCloseError": [MockFunction],
        "metaCreatePoll": [MockFunction],
        "metaDeletePoll": [MockFunction],
        "metaGetPolls": [MockFunction],
        "metaGetUser": [MockFunction],
        "metaNoOp": [MockFunction],
        "metaSetLoading": [MockFunction],
        "metaSetState": [MockFunction],
        "metaTimeoutError": [MockFunction],
        "pollAddOption": [MockFunction],
        "pollCastVote": [MockFunction],
        "pollRemoveOption": [MockFunction],
        "pollToggleFlag": [MockFunction],
        "pollToggleHide": [MockFunction],
        "pollToggleSecret": [MockFunction],
        "viewClearState": [MockFunction],
        "viewSetAdd": [MockFunction],
        "viewToggleConfirm": [MockFunction],
        "viewToggleSettings": [MockFunction],
      }
    }
    data={
      Object {
        "errors": Array [],
        "form": Object {
          "add": "",
          "confirm": false,
          "options": Array [],
          "secret": false,
          "title": "",
        },
        "list": Object {
          "search": "",
        },
        "loading": false,
        "polls": Array [],
        "user": Object {
          "auth": Object {
            "id": "",
            "provider": "",
          },
          "data": Object {
            "restricted": false,
          },
          "id": "id-a",
          "name": "",
          "type": "auth",
        },
        "view": Object {
          "add": "",
          "confirm": false,
          "settings": true,
        },
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    location={Object {}}
    match={Object {}}
    poll={
      Object {
        "author": "",
        "date": 0,
        "id": "",
        "options": Array [],
        "secret": false,
        "title": "",
        "users": Object {
          "created": "id-a",
          "flagged": Array [],
          "hidden": Array [],
          "voted": 0,
        },
      }
    }
    staticContext={Object {}}
  />
</div>
`;

exports[`view menu should match snapshot (default) 1`] = `
<div
  className="c-view-menu"
>
  <ViewControls
    actions={
      Object {
        "formAddOption": [MockFunction],
        "formClearState": [MockFunction],
        "formRemoveOption": [MockFunction],
        "formSetAdd": [MockFunction],
        "formSetTitle": [MockFunction],
        "formToggleConfirm": [MockFunction],
        "formToggleSecret": [MockFunction],
        "listClearState": [MockFunction],
        "listSetSearch": [MockFunction],
        "metaAddErrors": [MockFunction],
        "metaCloseError": [MockFunction],
        "metaCreatePoll": [MockFunction],
        "metaDeletePoll": [MockFunction],
        "metaGetPolls": [MockFunction],
        "metaGetUser": [MockFunction],
        "metaNoOp": [MockFunction],
        "metaSetLoading": [MockFunction],
        "metaSetState": [MockFunction],
        "metaTimeoutError": [MockFunction],
        "pollAddOption": [MockFunction],
        "pollCastVote": [MockFunction],
        "pollRemoveOption": [MockFunction],
        "pollToggleFlag": [MockFunction],
        "pollToggleHide": [MockFunction],
        "pollToggleSecret": [MockFunction],
        "viewClearState": [MockFunction],
        "viewSetAdd": [MockFunction],
        "viewToggleConfirm": [MockFunction],
        "viewToggleSettings": [MockFunction],
      }
    }
    data={
      Object {
        "errors": Array [],
        "form": Object {
          "add": "",
          "confirm": false,
          "options": Array [],
          "secret": false,
          "title": "",
        },
        "list": Object {
          "search": "",
        },
        "loading": false,
        "polls": Array [],
        "user": Object {},
        "view": Object {
          "add": "",
          "confirm": false,
          "settings": false,
        },
      }
    }
    history={
      Object {
        "push": [MockFunction],
      }
    }
    location={Object {}}
    match={Object {}}
    poll={
      Object {
        "author": "",
        "date": 0,
        "id": "",
        "options": Array [],
        "secret": false,
        "title": "",
        "users": Object {
          "created": "",
          "flagged": Array [],
          "hidden": Array [],
          "voted": 0,
        },
      }
    }
    staticContext={Object {}}
  />
</div>
`;
